website/content/en/docs/kubebuilder/references/client.md

```diff
@@ -0,0 +1,574 @@
+
+Example:
+
+```Go
+import (
+	"context"
+	"fmt"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	// Return all pods in the request namespace with a label of `app=<name>`
+	// and phase `Running`.
+	podList := &v1.PodList{}
+	opts := []client.ListOption{
+		client.InNamespace(request.NamespacedName.Namespace),
+		client.MatchingLabels{"app": request.NamespacedName.Name},
+		client.MatchingFields{"status.phase": "Running"},
+	}
+	ctx := context.TODO()
+	err := r.List(ctx, podList, opts...)
+
+	...
+}
+```
+
+[list-options]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#ListOptions
+[matching-labels]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#MatchingLabels
+[matching-fields]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#MatchingFields
+[in-namespace]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#InNamespace
+
+#### Create
+
+```Go
+// Create saves the object obj in the Kubernetes cluster.
+// Returns an error
+func (c Client) Create(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error
+```
+
+A `client.CreateOption` is an interface that sets [`client.CreateOptions`][create-options] fields. A `client.CreateOption` is created by using one of the provided implementations: [`DryRunAll`][dry-run-all], [`ForceOwnership`][force-ownership]. Generally these options are not needed.
+
+Example:
+
+```Go
+import (
+	"context"
+	"k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	app := &v1.Deployment{ // Any cluster object you want to create.
+		...
+	}
+	ctx := context.TODO()
+	err := r.Create(ctx, app)
+
+	...
+}
+```
+
+[create-options]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#CreateOptions
+
+#### Update
+
+```Go
+// Update updates the given obj in the Kubernetes cluster. obj must be a
+// struct pointer so that obj can be updated with the content returned
+// by the API server. Update does *not* update the resource's status
+// subresource
+func (c Client) Update(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error
+```
+
+A `client.UpdateOption` is an interface that sets [`client.UpdateOptions`][update-options] fields. A `client.UpdateOption` is created by using one of the provided implementations: [`DryRunAll`][dry-run-all], [`ForceOwnership`][force-ownership]. Generally these options are not needed.
+
+Example:
+
+```Go
+import (
+	"context"
+	"k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Update for new scaffold. `request reconcile.Request`==>`req ctrl.Request`
Update the imports as well `"sigs.k8s.io/controller-runtime/pkg/reconcile"`==>`ctrl "sigs.k8s.io/controller-runtime"`